app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -1,54 +1,142 @@
 import { SuggestionDb, groupColorStyle, type Group } from '@/stores/suggestionDatabase'
 import { tryGetIndex } from '@/util/array'
 import { Ast, AstExtended } from '@/util/ast'
+import { AliasAnalyzer } from '@/util/ast/aliasAnalysis'
 import { colorFromString } from '@/util/colors'
 import { ComputedValueRegistry, type ExpressionInfo } from '@/util/computedValueRegistry'
+import { MappedKeyMap } from '@/util/containers'
 import { ReactiveDb, ReactiveIndex, ReactiveMapping } from '@/util/database/reactiveDb'
 import type { Opt } from '@/util/opt'
 import { qnJoin, tryQualifiedName } from '@/util/qualifiedName'
 import { Vec2 } from '@/util/vec2'
+import { iteratorFilter } from 'lib0/iterator'
 import * as set from 'lib0/set'
 import {
   IdMap,
   visMetadataEquals,
+  type ContentRange,
   type ExprId,
   type NodeMetadata,
   type VisualizationMetadata,
 } from 'shared/yjsModel'
 import { ref, type Ref } from 'vue'
 
-export class GraphDb {
-  nodes = new ReactiveDb<ExprId, Node>()
-  idents = new ReactiveIndex(this.nodes, (_id, entry) => {
-    const idents: [ExprId, string][] = []
-    entry.rootSpan.visitRecursive((span) => {
-      if (span.isTree(Ast.Tree.Type.Ident)) {
-        idents.push([span.astId, span.repr()])
-        return false
+export interface AliasInfo {
+  identifier: AstExtended
```

Comment:
Let's not store `AstExtended` in here. Instead, store just the `ExprId`, and maybe identifier name if it helps with anything.